src/coreclr/vm/stubhelpers.cpp

```diff
@@ -498,96 +498,52 @@ extern "C" void QCALLTYPE StubHelpers_ThrowInteropParamException(INT resID, INT
 }
 
 #ifdef PROFILING_SUPPORTED
-FCIMPL3(SIZE_T, StubHelpers::ProfilerBeginTransitionCallback, SIZE_T pSecretParam, Thread* pThread, Object* unsafe_pThis)
+FCIMPL0(INT32, StubHelpers::ProfilerGetSystemError)
 {
     FCALL_CONTRACT;
+    return ::GetLastError();
```

Comment:
This is same as `Mashal.Get/SetLastSystemError`. Should we just use that instead of introducing these FCalls?

Reply:
Or preserve the last error in the FCall like it used to be

Reply:
> This is same as `Mashal.Get/SetLastSystemError`. Should we just use that instead of introducing these FCalls?

We can't since that will trigger a notification. I was preferring to keep the logic contained in the managed callout. These are very narrow FCalls and I'd prefer not to obscure them in the manner they were. We also no longer have a continuous FCall so I don't think I can use those macros. Perhaps I misunderstood their use. I'll revisit.

Reply:
@jkotas I missed your edit. We can store this in the QCall, but my concern is inadvertant adding of steps in the managed section that can update the last error. I've not audited the QCall paths to ensure they will never call into an API that updates last error on Windows or `errno` on non-Windows.

I'm happy to do the preservation in the QCall if you think that is the preferred pattern in this case.

Reply:
I think it would look better to have the save/restore of the last error in the QCall C++ code.

Also, I would emit a call to `Delegate.GetInvokeMethod` in the IL stub to get the MethodDesc* for the delegate case and pass it to the profiler callback. It would centralize the handling of the MethodDesc for the delegate case in one spot, it would not be spread over 3 different places.

(I do not have strong opinion about this.)

Reply:
Okay. Shouldn't too be hard to address.